crates/solver/src/liquidity/balancer_v2.rs

```diff
@@ -145,18 +144,18 @@ impl SettlementHandler {
         execution: AmmOrderExecution,
         encoder: &mut SettlementEncoder,
     ) -> Result<()> {
-        let (asset_in, amount_in) = execution.input;
+        let (asset_in, amount_in_max) = execution.input_max;
         let (asset_out, amount_out) = execution.output;
 
-        encoder.append_to_execution_plan(self.allowances.approve_token(asset_in, amount_in)?);
+        encoder.append_to_execution_plan(self.allowances.approve_token(asset_in, amount_in_max)?);
```

Comment:
This was a sneaky bug that we didn't run into thankfully! We were checking allowances with the `amount_in` before slippage, but we would transfer up to `amount_in + slippage` out - so we could potentially revert 😱